spec.html

```diff
@@ -188,6 +188,16 @@ <h1>The ElementDescriptor Specification Type</h1>
               <li>_element_.[[Descriptor]].[[Configurable]] is *false*.</li>
             </ul>
           </li>
+          <li>
+            If _element_.[[Kind]] is `"initializer", then
+            <ul>
+              <li>_element_.[[Key]] absent.</li>
+              <li>_element_.[[Descriptor]] is absent.</li>
+              <li>_element_.[[Placement]] is `"own"`.</li>
```

Comment:
We might consider allowing "static" as the decorator way of doing https://github.com/tc39/proposal-class-static-block

Reply:
I agree; static should be allowed here since they also can have initializers

Reply:
I don't see this as a way towards a static block. This is just a way that a *decorator* can add extra code to run. But decorators are already able to run code at the end of when a class is executing, in the finisher callback. We could allow this out of consistency (it would have slightly different timing from finishers), but I don't see a use case.

Reply:
A use case is the same as for "own" - being able to invoke a setter on a superclass (in this case, on a superclass's constructor that had, say, `static set foo(v) {}`)

Reply:
@ljharb What would be the problem with using a finisher for that?

Reply:
Certainly you could use a finisher for that - but then you'd be forced to use a class-level decorator to change the semantics of a specific field, so instead of:
```jsx
class C {
  @useSet static a = x;
  static b = y;
  @useSet static c = z;
}
```
you'd need to have:
```jsx
@useSet('a')
@useSet('c')
class C {
  static a = x;
  static b = y;
  static c = z;
}
```
or:
```jsx
@useSet('a', 'c')
class C {
  static a = x;
  static b = y;
  static c = z;
}
```
both of which are decidedly less ergonomic and elegant.

Reply:
I don't see why that would be the case. You can use a finisher on a static field declaration too.

Reply:
@littledan I have similar concern about `finisher` and initializer field. They kinda duplicate each other. E.g. if initializer fields can be `static` it almost removes profit finisher brings. If they cannot, it looks strange: why they cannot do it? 

Then I came to a thought: why not to do the `constructor` callback that works exactly as `finisher`, but during class instantiation? It would reduce not only amount of code that is necessary to describe the logic, but would also be clearer and obvious to use. Or am I missing something?

Reply:
@ljharb, wait, but you **can** use `finisher` with a field or a method decorator. It is not required to use class-level decorators to activate finishers.

Reply:
ahh, that wasn't clear to me. In that case - instead of this "initializer" approach, is there a reason not to allow finishers on "own" fields, to provide an "instance finisher"?

Reply:
@ljharb yeah, I came to the same idea in my [comment](https://github.com/tc39/proposal-decorators/pull/165#issuecomment-435520319) below :)